engineapi/backup_monitor.go

```diff
@@ -77,26 +102,91 @@ func NewBackupMonitor(logger logrus.FieldLogger,
 }
 
 func (m *BackupMonitor) monitorBackups() {
+	fallbackToExponentialBackoff := false
+
+	// Periodically liner timer
 	wait.PollUntil(BackupMonitorSyncPeriod, func() (done bool, err error) {
-		if err := m.syncBackups(); err != nil {
-			m.logger.Errorf("Stop monitoring because of %v", err)
+		fallback, err := m.syncBackups()
+		if err != nil && !fallback {
+			m.logger.Errorf("Stop monitoring because of %v in linear timer", err)
 			m.Close()
-			return false, err
+			return false, nil
+		}
+		if fallback {
+			fallbackToExponentialBackoff = fallback
+			return true, nil
 		}
 		return false, nil
 	}, m.ctx.Done())
+
+	if !fallbackToExponentialBackoff {
+		return
+	}
+
+	// Exponential back off timer
+	m.logger.Info("Fallback to exponential back off timer")
+	var (
+		initBackoff   = BackupMonitorSyncPeriod
+		maxBackoff    = 10 * time.Minute
+		resetDuration = BackupMonitorMaxRetryPeriod
```

Comment:
The `resetDuration` is for if the backoff is not called within it, the backoff is reset.
I was thinking we should not reset the backoff until we reach the BackupMonitorMaxRetryPeriod. (Which means it would never be reset).